### ISSUE TITLE
main contains outdated work

### ISSUE THREAD

**Issue by Rasbats** [opencpn-manuals/main#4]
The work in main should be taken from:

[https://github.com/opencpn-manuals/plugins](https://github.com/opencpn-manuals/plugins)

This contains work to fix the plugins manuals to a specific commit.

**Comment by leamas** [comment]
Ack. Looks like the stuff above comes from the experimental branch I had in ocpn-manuals, and we now need to move it back.

**Comment by leamas** [comment]
Just for the record: this is of course directly related to #2

**Comment by leamas** [comment]
I have merged the state branch into main in cd8f90b9. This solves the first part of #2 i. e., to use a defined set of sources using the source-state.sh script.

Time to close this bug?

**Comment by Rasbats** [comment]
This has not solved the problem in the plugins repo: 

[https://github.com/opencpn-manuals/plugins/issues/1](https://github.com/opencpn-manuals/plugins/issues/1)

Does sources.state.sh work for you in the main repo? 
And perhaps we need to look at sparse-checkout?

**Comment by leamas** [comment]
> Does sources.state.sh work for you in the main repo?

At a glance, yes. It does the basic operations restore, save and update seemingly ok. What's your problem with it?

**Comment by leamas** [comment]
> And perhaps we need to look at sparse-checkout?

I still don't really understand what it does and why it's useful. But locked into "finding the modules" right now.

**Comment by Rasbats** [comment]
I think the main problem here is that opencpn-manuals/plugins does not use an antora.yml in the root. 
Using playdo branches in main and plugins to test this.
Closing.

**Comment by leamas** [comment]
Perhaps not. After all, the antora job works and generates HTML. I have inspected the generated stuff, and it looks OK.

However, the workflow where we push main -> gh-pages and then updates gh-pages in the build job seems broken. The problem is that as soon as github  sees something pushed to gh-pages it will try to publish it. We have now pushed multiple commits on gh-pages whcih not are usable websites but rather the sources -- chances is that github has blacklisted us for now. In which case we should just refrain from pusing things to gh-pages for now.

It would be better to use an internediart branch, say ' build'. And that the automatic job takes input from 'build' and pushes the resulting website to gh-pages.

**Comment by leamas** [comment]
Yup. If I push something on  gh-pages in the plugins repo, nothing happens. OTOH, if I do it in the main repo things *does* happen. 

Could you please avoid  pushing things to gh-pages on  plugins? I think we need to wait for some github timeout before we can push again.

In the meantime, I will try to re-organize the workflow. And perhaps drop these utterl over-engineered upload, download and deployment tools.

**Comment by Rasbats** [comment]
Nothing pushed to gh-pages at this end.  Working locally with branches playdo in plugins and main.

The way you have added development to sources.state is what I have been trying to do for plugins. Now got antora.yml in opencpn-manuals/plugins but I think I may have to add a module to mimic development.

Idea (maybe incorrct) is to make it possible to use source-state.sh to extract/use a stable commit from opencpn-manuals/plugins.

**Comment by leamas** [comment]
I have pushed a single line commit to the ci branch. After this, one have to push to the 'build' branch top trigger a build i. e., something like `git push -f origin main:build`or so.

**Comment by leamas** [comment]
> Idea (maybe incorrct) is to make it possible to use source-state.sh to extract/use a stable commit from opencpn-manuals/plugins.

This is my "lock" idea (?) That is, checkout a specific version of sources/plugins. And the mark this is as "locked" i. e., that update does not affect it.

Likewise, there is always git. You could checking sources.state in a specific branch; this is the way to handle different version rather than keeping sources.bak etc around.

**Comment by leamas** [comment]
Some progress: the ci branch now actually publishes something. It's on https://opencpn-manuals.github.io/plugins/, 
and it's not nice. The local build looks fine, for sure. Extremely frustrating, this.

The javascript debugger says

```
Loading failed for the <script> with source “https://opencpn-manuals.github.io/plugins/_/js/site.js”. index.html:836:1
Loading failed for the <script> with source “https://opencpn-manuals.github.io/plugins/_/js/vendor/highlight.js index.html:837:1 ”.
```

Sigh.

**Comment by leamas** [comment]
Yes, something is broken with the access to site.js. It is probably available at the site, if I pull gh-pages I can see it in the proper location _/js/site.js.

The urls to site.js is  https://opencpn-manuals.github.io/plugins/_/js/site.js, and cannot be accessed.

For the main repo, the corresponding url is  https://opencpn-manuals.github.io/main/_/js/site.js, and this works fine.

**Comment by rgleason** [comment]
that looks like nojeykll problems I was having.

**Comment by leamas** [comment]
Indeed. .nojekyll is available in the working  main repo, but not in plugins. And what this  default github pages processing is to remove the directory named _, which is where the javascript files lives.

**Comment by leamas** [comment]
@rasbats. At last, the ci branch now also publishes results correctly. Results available at https://opencpn-manuals.github.io/plugins/
and looks reasonable. I suggest that you merge the ci branch into main if you don't see any obvious problems

@rgleason : Thanks.

If/when the ci branch is merged I guess we could close this issue?

**Comment by leamas** [comment]
I have rebased ci on main to simplify the merge.

**Comment by Rasbats** [comment]
main(playdo branch) has slimmed down to sources and modules. Most of the other stuff has been removed. Problems due to source-state.sh with update but locally I can make a build with everything in sources.

Site.yml has to have a list of plugin sources to link with the dev plugin repos in sources but perhaps that could be scripted. Easy to add a new plugin.

Ok ... enough for today.

**Comment by leamas** [comment]
I have pushed my work on top of playdo. It now works also when publishing. Dropped the complicated npm modules and made a simple script which is way faster (no upload/download of site) and much simpler to configure

The problems you had with sources was since there was some (2) directories in sources which were checked-in in git. This creates conflicts with source-state.sh. Fixed.

In  order to not accidentally check in stuff in sources it must be listed in .gitignore. Fixed.

Please do not remove sources/.gitkeep. A .gitkeep file is the usual way to check in an otherwise empty directory into git. We need the sources directory, but you cannot check in a directory in git. Hence the .gitkeep file which basically ensures that sources exists.

Yes, we might need to script the handling of sources in site.yml somehow. OTOH, we don't add or remove plugins that often, so it's probably a low priority for now (?)

Thanks for your work, and for good commit comments :)

**Comment by Rasbats** [comment]
Cloning into 'sar_pi'...
remote: Enumerating objects: 53, done.
remote: Counting objects: 100% (53/53), done.
remote: Compressing objects: 100% (40/40), done.
remote: Total 53 (delta 1), reused 39 (delta 0), pack-reused 0
Receiving objects: 100% (53/53), 14.08 KiB | 2.01 MiB/s, done.
Resolving deltas: 100% (1/1), done.
remote: Enumerating objects: 13, done.
Any reason for this error  when running source-state.sh? Despite this the sources are built.
```
remote: Counting objects: 100% (13/13), done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 13 (delta 0), reused 10 (delta 0), pack-reused 0
Receiving objects: 100% (13/13), 39.75 KiB | 1.53 MiB/s, done.
remote: Enumerating objects: 36, done.
remote: Counting objects: 100% (36/36), done.
remote: Compressing objects: 100% (36/36), done.
remote: Total 36 (delta 0), reused 35 (delta 0), pack-reused 0
Receiving objects: 100% (36/36), 1.09 MiB | 3.92 MiB/s, done.
Updating files: 100% (49/49), done.
fatal: invalid refspec '543af5d0bcbf8024fab8be15b3fb1fe3a140b721?'
error: pathspec 'FETCH_HEAD' did not match any file(s) known to git
```

**Comment by leamas** [comment]
Doesn't happen for me (after fixing things). Have you pulled my changes?

**Comment by leamas** [comment]
... ant it doesn't happen in the [action workflow](https://github.com/opencpn-manuals/main/runs/3296786501?check_suite_focus=true)

**Comment by Rasbats** [comment]
Thanks for the fixes.
Yes ... fresh clone. Annoying but the script still does the business.

**Comment by leamas** [comment]
You should try to just remove sources/sar_pi completely before running restore.

**Comment by Rasbats** [comment]
Not just sar_pi with the same error. A number.
Anyway ... spending the morning moving the user manual to asciidoc. Hope no-one has already done this.

**Comment by leamas** [comment]
> Not just sar_pi with the same error. A number.

Just remove everything under sources/ and run restore. Should fix it.

> Anyway ... spending the morning moving the user manual to asciidoc. Hope no-one has already done this.

Unless Rick has, which I doubt. Isn't  this a huge undertaking?

**Comment by leamas** [comment]
BTW: Before you dive into the user manual: Isn't it time to merge playdo into main?

**Comment by Rasbats** [comment]
>Isn't it time to merge playdo into main?

Yes but I get conflicts:
```
Removing dev/.nojekyll
Removing dev/.gitignore
CONFLICT (add/add): Merge conflict in ci/build-and-deploy.sh
Auto-merging ci/build-and-deploy.sh
Removing build.yml
Removing beta_plugins.txt
CONFLICT (add/add): Merge conflict in INSTALL.md
Auto-merging INSTALL.md
Auto-merging .github/workflows/main.yml
CONFLICT (content): Merge conflict in .github/workflows/main.yml
Automatic merge failed; fix conflicts and then commit the result.

C:\Projects\opencpn-manuals\main>
```
Would appreciate some help resolving these. No good with conflicts.

**Comment by leamas** [comment]
OK, I'll take care of it. No worries...

**Comment by leamas** [comment]
... done. The build jobs for main repo are stuck since 10 hours, so nothing is rebuilt to github pages. These things tends to resolve over time (github builders doesn't seem to be that stable, BTW).


Time to close this bug?

**Comment by leamas** [comment]
NO, it's not. I can reproduce your failiing sar_pi. Will look into it.

**Comment by leamas** [comment]
Fixed. It was not about sar_pi, it was about the development manual source. Pushed a quick fix for now in main branch, will try to clean it up later.

Build log [here](https://github.com/opencpn-manuals/main/actions/runs/1119321088), results at https://opencpn-manuals.github.io/main/

Time to close this bug?

**Comment by Rasbats** [comment]
Yes ...

**Comment by leamas** [comment]
I'll drop the merged branches later, to keep things sane...
